docs/DeveloperGuide.md

```diff
@@ -4,26 +4,207 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+# 1. Overview
+MeetingJio is a program for **finding potential time slots for team meetings based on everyone’s availability**.
+
+# 2. Design 
+
+## 2.1 Architecture
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## 2.2 UI Component
+
+## 2.3 Logic Component
+- Parsers
+- Commands
+
+## 2.4 Model Component
+- Timetables
+- Events
+
+## 2.5 Storage Component
+
+# 3. Implementation
+This section describes some noteworthy details on how certain features are implemented.
+
+## 3.1 Add Feature
+The `add` command is a command that allows user to 
+1. add new users to the master timetable.
+2. add new events to the timetables.
+
+The following sequence diagram shows how the `add` operation generally works.
+
+![AddCommandSequenceDiagram](diagrams/AddCommandSequenceDiagram.png)
+
+**_Note:_**
+
+This is a condensed diagram. Several terms in the sequence diagram above have been substituted by a common term.
+
+Common Term | User-specific term | Lesson-specific term | Meeting-specific term      |
+------------|-------------------|----------------------|----------------------------|        
+add | `"add_user ..."`  | `"add_lesson ..."`   | `"add_meeting ..."`     
+AddCommand | `AddUserCommand`  | `AddLessonCommand`   | `AddMeetingCommand`  
+
+
+## 3.1.1 Add User Feature
+`add_user` adds new user and his or her timetable into the master timetable.
+
+The following sequence diagram shows how the `add_user` command works:
+![AddUserCommandSequenceDiagram](diagrams/AddUserCommandSequenceDiagram.png)
```

Comment:
Consider using 

`[ -> Command` instead of `none -> Command`.

![image](https://user-images.githubusercontent.com/26570382/161184718-bca93ae1-4288-47dd-8def-6c9f939e6fd4.png)

This can help you remove the "Unspecified" object.